### ISSUE TITLE
Access tables in a consistent order when inside a transaction

### ISSUE THREAD

**Issue by chilts** [mozilla/fxa-auth-db-server#101]
Moving the discussion from https://github.com/mozilla/fxa-auth-server/issues/785 since the deadlock is related to the db server.

**Comment by rfk** [comment]
(I'm morphing this bug into a concrete task)

As noted in https://github.com/mozilla/fxa-auth-server/issues/785#issuecomment-62644895 it's generally a good idea, where possible, to have all your tranasctions touch tables in a consistent order.  This reduces the possibility of deadlocks due to two threads waiting on a table lock held by the other.

It seems like a pretty small change that could reduce a small-but-definitely-there source of errors in production.  Super low priority though.

**Comment by rfk** [comment]
closing in favour of issue inside auth-db-mysql repo
